src/server/browserContext.ts

```diff
@@ -195,9 +195,13 @@ export abstract class BrowserContext extends SdkObject {
     if (!this.pages().length) {
       const waitForEvent = helper.waitForEvent(progress, this, BrowserContext.Events.Page);
       progress.cleanupWhenAborted(() => waitForEvent.dispose);
-      await waitForEvent.promise;
+      const page = (await waitForEvent.promise) as Page;
+      if (page._pageIsError)
+        throw page._pageIsError;
```

Comment:
consider creating a new error and chaining it with page init error for better stack trace